mitodl/mitxonline
# fix: protect Product model from deletion

#### Pre-Flight checklist


#### What are the relevant tickets?
#732

#### What's this PR do?
- Disables the delete option for the products model through Admin
- Introduce Active Manager for products
- Introduce (Disable delete through Django Shell too)
- Adds db level protection for the Product model
- The deletion through Django would result in marking the product as in-active.

#### How should this be manually tested?
- You should not be able to delete product objects through Django Admin
- You should not be able to delete Products through Django Shell via `Product(instance).delete()` or `Product.objects.all().delete()` - try with all/filter
- Upon deleting a product it should be marked as inactive instead
- Go to db shell `./manage.py dbshell` and try to delete a product with SQL e.g. (`delete from ecommerce_product where id=1;`, the record should not be deleted)

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)
